### ISSUE TITLE
Remove gap in Highcharts column chart from null value.

### ISSUE THREAD

**Issue by luciomcamargo** [highcharts/highcharts-react#77]
I'd like to justify columns by category on a Highcharts column chart, so it corrects null values. The function below was written by Grzegorz does just that. But when I try to add the function in the React official wrapper, it returns "element" undefined. How do I adapt this to the React official wrapper. Remember I only want to ignore the gaps from null values, not zeros. Here is a codesandbox of my code, the logic should be in the Chart.js component. [https://codesandbox.io/s/github/luciomcamargo/graph](url)

```
var justifyColumns = function (chart) {
    var categoriesWidth = chart.plotSizeX / (1 + chart.xAxis[0].max - chart.xAxis[0].min),
        distanceBetweenColumns = 0,
        each = Highcharts.each,
        sum, categories = chart.xAxis[0].categories,
        number;
    for (var i = 0; i < categories.length; i++) {
        sum = 0;
        each(chart.series, function (p, k) {
            if (p.visible) {
                each(p.data, function (ob, j) {
                    if (ob.category == categories[i]) {
                        sum++;
                    }
                });
            }
        });
        distanceBetweenColumns = categoriesWidth / (sum + 1);
        number = 1;
        each(chart.series, function (p, k) {
            if (p.visible) {
                each(p.data, function (ob, j) {
                    if (ob.category == categories[i]) {
                        ob.graphic.element.x.baseVal.value = i * categoriesWidth + distanceBetweenColumns * number - ob.pointWidth / 2;
                        number++;
                    }
                });
            }
        });
    }
};




$(function () {
    $('#container').highcharts({
        chart: {
            type: 'column',
            animation: false,
            events: {
                load: function () {
                    justifyColumns(this);
                },
                redraw: function () {
                    justifyColumns(this);
                }
            }
        },
        title: {
            text: 'Monthly Average Rainfall'
        },
        subtitle: {
            text: 'Source: WorldClimate.com'
        },
        xAxis: {
            categories: [
                'Jan',
                'Feb',
                'Mar'],
            crosshair: true
        },
        yAxis: {
            min: 0,
            title: {
                text: 'Rainfall (mm)'
            }
        },
        tooltip: {
            headerFormat: '<span style="font-size:10px">{point.key}</span><table>',
            pointFormat: '<tr><td style="color:{series.color};padding:0">{series.name}: </td>' +
                '<td style="padding:0"><b>{point.y:.1f} mm</b></td></tr>',
            footerFormat: '</table>',
            shared: true,
            useHTML: true
        },
        plotOptions: {
            column: {
                borderWidth: 0
            },
            series: {
                events: {
                    hide: function () {
                        justifyColumns(this.chart);
                    },
                    show: function () {
                        justifyColumns(this.chart);
                    }
                }
            }
        },
        series: [{
            name: 'Tokyo',
            data: [
                [0, 99.9],
                [1, 71.5],
                [2, 106.4]
            ]

        }, {
            name: 'New York',
            data: [
                [0, 49.9],
                [2, 106.4]
            ]

        }, {
            name: 'London',
            data: [
                [0, 49.9],
                [1, 71.5],
                [2, 106.4]
            ]

        }, {
            name: 'Berlin',
            data: [
                [0, 49.9],
                [1, 71.5],
                [2, 106.4]
            ]

        }]
    });
});

```

**Comment by ppotaczek** [comment]
Hi luciomcamargo,

We try to keep GitHub only for bugs, so please ask this type of questions on support: https://www.highcharts.com/support

The answer to your question is on stackoverflow, in the thread you created:

https://stackoverflow.com/questions/53697672/remove-gap-in-highcharts-column-chart-from-null-value-from-json-file-in-react/53705100#53705100

Best regards!

**Comment by luciomcamargo** [comment]
Hi,  I really appreciate the support.  But the effect I desired was to join
the last columns together,  so groupPadding 0, but completely ignoring the
null value on the chart.  I wrote a comment in the stackoverflow thread.

Regards,

Lucio

**Comment by luciomcamargo** [comment]
Hello Pawel,

I'm really sorry to bother you again, but I needed to submit this by
tomorrow morning for a job application. I posted on stackoverflow as well.
I'm uploading an image of what I want to achieve so it's clear. I tried to
clarify in the comments as well. The only reason I used that justify
function was because I was trying eliminate the last gap. Thank you for the
support.

Regards,

Lucio
